staging/src/k8s.io/legacy-cloud-providers/aws/aws.go

```diff
@@ -1202,7 +1203,13 @@ func azToRegion(az string) (string, error) {
 	if len(az) < 1 {
 		return "", fmt.Errorf("invalid (empty) AZ")
 	}
-	region := az[:len(az)-1]
+
+	r := regexp.MustCompile(`\w+-(gov-)?\w+-\d+`)
```

Comment:
this is unsafe.
e.g. "us-iso-east-1" gives "iso-east-1", also we need to at least include "^" as first character of this regex.

You can find all possible region regex currently by search "regionRegex" in https://github.com/boto/botocore/blob/master/botocore/data/endpoints.json

Reply:
BTW, seems a `^([a-zA-Z]+-)+\d+` will work, but better to confirm with team in region build.

Reply:
Em. if `iso` is another case we need catch, then `\w+-(gov-|iso-)?\w+-\d+` will work. or just use simple `^([a-zA-Z]+-)+\d+` or `^([a-zA-Z]+-)+\d` and assume there won't be numeric value in the middle.

Reply:
BTW, regionRegex `^(us|eu|ap|sa|ca|me|af)\\-\\w+\\-\\d+$` can not cover local zone in this senario, it has to be `[a-zA-z]` instead of `\w`.